Prox/Prox/Utilities/CategoriesUtil.swift

```diff
@@ -124,6 +124,33 @@ struct CategoriesUtil {
         }
         return categoryToName
     }()
+
+    static let categoryToFilter: [String: Filter] = {
+        var categoryToFilter = [String: Filter]()
+
+        // Build the initial map of categories -> filters that we set explicitly.
+        for (filter, categories) in Filter.categories {
+            for category in categories {
+                categoryToFilter[category] = filter
+            }
+        }
+
+        // Set the remaining categories based on the first ancestor with an explicitly-set filter.
+        for category in categoryToParentsMap.keys {
+            guard let firstCategorizedParent = getAncestors(forCategory: category).flatMap({ categoryToFilter[$0] }).first else { continue }
```

Comment:
Is it ever expected that a category will not have a categorized ancestor? If so, explain why not (on the outer `categoryToFilter` var?). If not, we should crash (since this should be caught by tests. Also, write a test :) ).